eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -96,7 +97,34 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, router: ActorRef
       require(remoteNodeId == remoteNodeId1, s"invalid nodeid: $remoteNodeId != $remoteNodeId1")
       log.debug(s"got authenticated connection to $remoteNodeId@${address.getHostString}:${address.getPort}")
 
-      context watch peerConnection
+      // We want to log all incoming/outgoing messages to/from this peer. Logging incoming messages is easy (they all go
+      // through this actor), but for outgoing messages it's a bit trickier because channels directly send messages to
+      // the connection. That's why we use this pass-through actor that just logs messages and forward them.
+      val logConnection = context.actorOf(Props(new Actor {
+        // we use this to log raw messages coming in and out of the peer
+        val logMsgOut = new BusLogging(context.system.eventStream, "", classOf[Peer.MessageLogs], context.system.asInstanceOf[ExtendedActorSystem].logFilter) with DiagnosticLoggingAdapter
+        context watch peerConnection
+        override def receive: Receive = {
+          case Terminated(_) =>
+            context stop self
+          case msg =>
+            msg match {
+              case _: LightningMessage =>
+                logMsgOut.mdc(mdc(msg))
```

Comment:
Is the inner call to `mdc(msg)` really safe (since DiagnosticActorLogging explicitly mentions that it's not thread-safe)? It feels like the Peer's `mdc` function can be executing in two different threads at the same time if we're processing messages in both the `Peer` and `PeerConnection` concurrently (which we will).

And how would that captured lambda even work when a Peer and its PeerConnection end up on different hosts?

Reply:
Setting/clearing the MDC isn't thread because it uses thread-local storage, but the `def mdc(currentMessage: Any): MDC` method is threadsafe, it's a simple stateless method that just returns a new immutable Map.

Reply:
> And how would that captured lambda even work when a Peer and its PeerConnection end up on different hosts?

Not sure what you mean

Reply:
Ok I got confused there thinking it would somehow be stateful and would ruin everything. This can probably be ignored, your answers make sense.